src/transformState.js

```diff
@@ -5,7 +5,32 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const obj of actions) {
+    switch (obj.type) {
+      case 'addProperties':
+        Object.assign(state, { ...obj.extraData });
+        break;
+
+      case 'removeProperties':
+        const deleteData = obj.keysToRemove;
+
+        for (const deleteKey of deleteData) {
+          delete state[deleteKey];
+        }
+        break;
+
+      case 'clear':
+        for (const key in state) {
+          delete state[key];
+        }
+        break;
+
+      default:
+        return false;
```

Comment:
Instead of returning false, you should continue to the next iteration when an unknown action is encountered.

Reply:
or throw the error